execution/gethexec/sync_monitor.go

```diff
@@ -137,3 +144,21 @@ func (s *SyncMonitor) BlockMetadataByNumber(blockNum uint64) (common.BlockMetada
 	log.Debug("FullConsensusClient is not accessible to execution, BlockMetadataByNumber will return nil")
 	return nil, nil
 }
+
+func (s *SyncMonitor) StoreFinalityData(ctx context.Context, finalityData *arbutil.FinalityData) error {
+	s.finalityData.Store(finalityData)
+
+	finalizedBlockNumber, err := s.FinalizedBlockNumber(ctx)
+	if errors.Is(err, headerreader.ErrBlockNumberNotSupported) {
+		log.Warn("Finality not supported so not setting finalized block number")
+	} else if err == nil {
+		err = s.exec.SetFinalized(finalizedBlockNumber)
```

Comment:
let's also use blockcahin's SetSafe while we're at it.
We can have either a single function in exec calling both or two funcstions, whichever seems better.